pkg/operation/botanist/controlplane/kubescheduler/kube_scheduler.go

```diff
@@ -0,0 +1,473 @@
+								LocalObjectReference: corev1.LocalObjectReference{
+									Name: configMap.Name,
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (k *kubeScheduler) SetSecrets(secrets Secrets)          { k.secrets = secrets }
+func (k *kubeScheduler) Destroy(_ context.Context) error     { return nil }
+func (k *kubeScheduler) Wait(_ context.Context) error        { return nil }
+func (k *kubeScheduler) WaitCleanup(_ context.Context) error { return nil }
+
+func (k *kubeScheduler) emptyConfigMap() *corev1.ConfigMap {
+	return &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: "kube-scheduler-config", Namespace: k.namespace}}
+}
+
+func (k *kubeScheduler) emptyVPA() *autoscalingv1beta2.VerticalPodAutoscaler {
+	return &autoscalingv1beta2.VerticalPodAutoscaler{ObjectMeta: metav1.ObjectMeta{Name: "kube-scheduler-vpa", Namespace: k.namespace}}
+}
+
+func (k *kubeScheduler) emptyService() *corev1.Service {
+	return &corev1.Service{ObjectMeta: metav1.ObjectMeta{Name: "kube-scheduler", Namespace: k.namespace}}
+}
+
+func (k *kubeScheduler) emptyDeployment() *appsv1.Deployment {
+	return &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: v1beta1constants.DeploymentNameKubeScheduler, Namespace: k.namespace}}
+}
+
+func (k *kubeScheduler) computeServerPort() int32 {
+	if k.versionConstraints.k8sGreaterEqual113 {
+		return 10259
+	}
+	return 10251
+}
+
+func (k *kubeScheduler) computeServerURIScheme() corev1.URIScheme {
+	if k.versionConstraints.k8sGreaterEqual113 {
+		return corev1.URISchemeHTTPS
+	}
+	return corev1.URISchemeHTTP
+}
+
+func (k *kubeScheduler) computeEnvironmentVariables() []corev1.EnvVar {
+	if k.config != nil && k.config.KubeMaxPDVols != nil {
+		return []corev1.EnvVar{{
+			Name:  "KUBE_MAX_PD_VOLS",
+			Value: *k.config.KubeMaxPDVols,
+		}}
+	}
+	return nil
+}
+
+func (k *kubeScheduler) computeComponentConfig() (string, string, error) {
+	var apiVersion string
+	if k.versionConstraints.k8sGreaterEqual118 {
+		apiVersion = "kubescheduler.config.k8s.io/v1alpha2"
+	} else if k.versionConstraints.k8sGreaterEqual112 {
+		apiVersion = kubeschedulerv1alpha1.SchemeGroupVersion.String()
+	} else if k.versionConstraints.k8sGreaterEqual110 {
+		apiVersion = "componentconfig/v1alpha1"
+	}
+
+	componentConfig := &kubeschedulerv1alpha1.KubeSchedulerConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: apiVersion,
+			Kind:       "KubeSchedulerConfiguration",
+		},
+		ClientConnection: componentbaseconfigv1alpha1.ClientConnectionConfiguration{
+			Kubeconfig: volumeMountPathKubeconfig + "/" + secrets.DataKeyKubeconfig,
+		},
+		LeaderElection: kubeschedulerv1alpha1.KubeSchedulerLeaderElectionConfiguration{
+			LeaderElectionConfiguration: componentbaseconfigv1alpha1.LeaderElectionConfiguration{
+				LeaderElect: pointer.BoolPtr(true),
+			},
+		},
+	}
+
+	componentConfigYAML, err := NewConfigCodec().Encode(componentConfig)
```

Comment:
I would inject the codec, instead of creating it here. This would allow for it to be easily mocked away in tests, to be replaced by a different implementation, etc.

Reply:
Thanks, I think we will go back to a template for the configuration (at least for the kube-scheduler), see https://github.com/gardener/gardener/pull/2731#discussion_r475690713.